### PR TITLE
Fix/deprecation

### PR DESCRIPTION
**Pull request by asturio**
Here are some changes concerning old java code. I didn't change too much, because there is really a lot to do, but this is a beginning. You will get a better readable code using this "new" features.

### CODE REVIEW

**Comment by andreasrosdal** [comment]
@tlxtellef  This pull-request might be worth merging before making a new release.

**Comment by asturio** [comment]
@tlxtellef: Just pulled from master, to have less diffs. I think even if I gain write access, the changes should be reviewed by other developers.

File: README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+This is just a "private" copy. Please get OpenPDF from LibrePDF/OpenPDF.
+
```
Comment (asturio) [inline README.md]: I should delete this, sorry.

Comment (andreasrosdal) [inline README.md]: Do you really want this change?

**Comment by asturio** [comment]
@andreasrosdal  "private" notice is out.

File: openpdf/src/main/java/com/lowagie/text/Document.java [hunk 2]
```diff
@@ -126,7 +126,7 @@
     public static float wmfFontCorrection = 0.86f;
     
 	/** The DocListener. */
-    private ArrayList listeners = new ArrayList();
+    private java.util.List<DocListener> listeners = new ArrayList<>();
```
Comment (tlxtellef) [inline openpdf/src/main/java/com/lowagie/text/Document.java]: Seems like a `import java.util.List` is missing.

File: openpdf/src/main/java/com/lowagie/text/pdf/PdfAppearance.java [hunk 3]
```diff
@@ -55,8 +57,8 @@
  */
 
 public class PdfAppearance extends PdfTemplate {
-    
-    public static final HashMap stdFieldFontNames = new HashMap();
```
Comment (tlxtellef) [inline openpdf/src/main/java/com/lowagie/text/pdf/PdfAppearance.java]: Removing public access might (we don't know) break someones setup. Meaning if we include such changes it should by Semantic versioning be a Major version change. But, when something that should be internal was `public` by accident is fixed I'm so not sure.

File: openpdf/src/main/java/com/lowagie/text/pdf/PdfDocument.java [hunk 4]
```diff
@@ -1920,7 +1920,7 @@ void calculateOutlineCount() {
      * Recursive method to update the count in the outlines.
      */
     void traverseOutlineCount(PdfOutline outline) {
-        ArrayList kids = outline.getKids();
+        java.util.List<PdfOutline> kids = outline.getKids();
```
Comment (tlxtellef) [inline openpdf/src/main/java/com/lowagie/text/pdf/PdfDocument.java]: Seems like a `import java.util.List` is missing.

File: openpdf/src/main/java/com/lowagie/text/pdf/PdfAppearance.java (hunk 3 above)
Comment (asturio) [reply openpdf/src/main/java/com/lowagie/text/pdf/PdfAppearance.java]: Oh, I Understand. I'll revert th "public" removals. That is a good point.

File: openpdf/src/main/java/com/lowagie/text/pdf/PdfDocument.java (hunk 4 above)
Comment (asturio) [reply openpdf/src/main/java/com/lowagie/text/pdf/PdfDocument.java]: Not really. There is already an import for com.lowagie.text.List there, so we need here the FQN.

**Comment by asturio** [comment]
Hi @tlxtellef, my IDE reorganized too much methods. I'm doing some changes to minimize the diff.

File: openpdf/src/main/java/com/lowagie/text/pdf/PdfAppearance.java (hunk 3 above)
Comment (tlxtellef) [reply openpdf/src/main/java/com/lowagie/text/pdf/PdfAppearance.java]: We can take all of these issues that are breaking changes in the same release as the package renaming needed for Maven Central.

Comment (asturio) [reply openpdf/src/main/java/com/lowagie/text/pdf/PdfAppearance.java]: I think the change was really wrong. If the method was public, then it's part of the API, should be documented and there should also be a test for that, so the method will be used. But there is a thing about writing tests for legacy code :-( . Anyway, I'd like to generify the code more.

File: openpdf/src/main/java/com/lowagie/text/pdf/PdfOutline.java [hunk 5]
```diff
@@ -268,9 +268,9 @@ public PdfOutline(PdfOutline parent, PdfAction action, Paragraph title) {
      */
     public PdfOutline(PdfOutline parent, PdfAction action, Paragraph title, boolean open) {
         super();
-        StringBuffer buf = new StringBuffer();
-        for (Iterator i = title.getChunks().iterator(); i.hasNext(); ) {
-            Chunk chunk = (Chunk) i.next();
+        StringBuilder buf = new StringBuilder();
+        for (Object o : title.getChunks()) {
```
Comment (tlxtellef) [inline openpdf/src/main/java/com/lowagie/text/pdf/PdfOutline.java]: The next iteration of changes should be to add proper return types to things like `Phrase.getChunks()`, removing the need for this cast.

**Pull request merged**